sway-core/src/semantic_analysis/ast_node/expression/typed_expression/method_application.rs

```diff
@@ -324,15 +324,30 @@ fn re_parse_expression(
         }
     };
 
-    let contract_address = check!(
+    let span = Span {
+        span: contract_pair.as_span(),
+        path: Some(build_config.path()),
+    };
+    // contract_address_result should be empty I think?
+    let contract_address_result = check!(
         Expression::parse_from_pair(contract_pair, Some(build_config)),
         return err(warnings, errors),
         warnings,
         errors
     );
+    if !contract_address_result.var_decls.is_empty() {
```

Comment:
what if there's a match inside of the abi cast? like

`abi(SomeAbi, match x { })`? 

If we decide we don't want to support that, I suggest we make a specific error for it instead of using `ParseError`.

Reply:
They will have already been lifted during parsing. However admitting an error here is not the right thing to do and instead any resulting var_decls should be ignored.